lib/file_store/s3_store.rb

```diff
@@ -87,7 +87,7 @@ def store_file(file, path, opts = {})
       # cache file locally when needed
       cache_file(file, File.basename(path)) if opts[:cache_locally]
       options = {
-        acl: opts[:private_acl] ? "private" : "public-read",
+        acl: SiteSetting.s3_use_acls ? (opts[:private_acl] ? "private" : "public-read") : nil,
```

Comment:
I'm not sure how this fits with secure uploads, on paper, disabling ACLs looks like will also disable secure uploads. Maybe `s3_use_acls` should be a hidden setting to avoid admins shooting themselves in the foot? Not sure. 

Thoughts @martin-brennan? 

Curious what you think about this change, too, @andrewschleifer.

Reply:
Yes, this is absolutely better in the most common case.

Re: secure uploads -- with the caveat that I have only just read the manifesto for the first time -- I think they are currently incompatible but could be made to work:
1. write a bucket policy that makes only the necessary prefixes public -- like `assets/`
2. secure uploads can skip the ACL munging and just use the signed URLs

(A mistake in that first step would completely gut the security, though.)

Reply:
@pmusaraj you are right about the secure uploads, I will give this a closer look this week. My first thought was to just add a site setting validator for this scenario. If indeed this is the wave of the future, it would be ideal to not use the ACLs for secure uploads, but you are right in that I don't have bandwidth for this, and I've already had months of secure uploads work taken off my plate internally because of this :)

Reply:
> If indeed this is the wave of the future

I'd say it's the way of the present, even, since new S3 buckets have ACLs disabled by default, and you need to click through a couple of "are you *really* sure you want to do this?  We don't recommend it" modals to enable them in the console.

For example, at bucket creation time, you get this:

![bucket_creation](https://github.com/discourse/discourse/assets/357/22808a85-7c32-47d4-86c2-a5bdc19eb705)

If you *do* proceed to click the "ACLs enabled" button, you get an extra chance to rethink the folly of your ways:

![bucket_creation2](https://github.com/discourse/discourse/assets/357/b7e8a0d6-6f40-4051-a038-24b4ca2466af)

Modifying an existing bucket to use ACLs involves going to the "Object Ownership" heading, which isn't very obvious (although the description mentions ACLs, so a text search finds it easily enough).  If you hit the "Edit" button and then select "ACLs enabled", you get the same warning as previously, *plus* an extra one, just for good luck, along with a bonus checkbox you've got to tick to make sure you *really* understand:

![bucket_edit](https://github.com/discourse/discourse/assets/357/a198becf-2aee-4461-8007-2e9e231c4115)

There's also [the docs](https://docs.aws.amazon.com/AmazonS3/latest/userguide/ensure-object-ownership.html) that say:

> We recommend that you disable ACLs on your Amazon S3 buckets.

AWS even went so far as to [publish a blog post](https://aws.amazon.com/about-aws/whats-new/2022/12/amazon-s3-automatically-enable-block-public-access-disable-access-control-lists-buckets-april-2023/) announcing that they were changing the default for all new buckets to disable ACLs.  Given how fanatical AWS are about not breaking backwards compatibility, the fact that they changed this default shows they're in srs bznz mode.

All up, I'd say that AWS would *really* prefer it if you didn't use ACLs, and stick to resource policies instead.

> A mistake in that first step would completely gut the security, though.

Yeah, but that's AWS security all over.  It's [like the PM in The Dish says](https://www.youtube.com/watch?v=XSxl3OAwxps): "We've got a saying in the Party: you don't fuck up..."  Attach the wrong ACL and you'd be just as stuffed.